vscode/src/workspace.ts

```diff
@@ -56,7 +54,17 @@ export class Workspace implements WorkspaceInterface {
     }
 
     try {
-      await fs.access(this.workspaceFolder.uri.fsPath, fs.constants.W_OK);
+      const stat = await vscode.workspace.fs.stat(this.workspaceFolder.uri);
+      const permissions = stat.permissions;
```

Comment:
Nitpick: Do we really need this local variable? I guess we could just check for `stat.permissions` inside the `if` clause below, right?

Reply:
Yeah, that's better.